packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -330,16 +366,109 @@ interface TooltipContentProps extends TooltipContentImplProps {
 const TooltipContent = React.forwardRef<TooltipContentElement, TooltipContentProps>(
   (props: ScopedProps<TooltipContentProps>, forwardedRef) => {
     const portalContext = usePortalContext(CONTENT_NAME, props.__scopeTooltip);
-    const { forceMount = portalContext.forceMount, ...contentProps } = props;
+    const { forceMount = portalContext.forceMount, side = 'top', ...contentProps } = props;
     const context = useTooltipContext(CONTENT_NAME, props.__scopeTooltip);
+
     return (
       <Presence present={forceMount || context.open}>
-        <TooltipContentImpl ref={forwardedRef} {...contentProps} />
+        {context.disableHoverableContent ? (
+          <TooltipContentImpl side={side} {...contentProps} ref={forwardedRef} />
+        ) : (
+          <TooltipContentHoverable side={side} {...contentProps} ref={forwardedRef} />
+        )}
       </Presence>
     );
   }
 );
 
+type Point = { x: number; y: number };
+type Polygon = Point[];
+
+type TooltipContentHoverableElement = TooltipContentImplElement;
+interface TooltipContentHoverableProps extends TooltipContentImplProps {}
+
+const TooltipContentHoverable = React.forwardRef<
+  TooltipContentHoverableElement,
+  TooltipContentHoverableProps
+>((props: ScopedProps<TooltipContentHoverableProps>, forwardedRef) => {
+  const context = useTooltipContext(TRIGGER_NAME, props.__scopeTooltip);
+  const providerContext = useTooltipProviderContext(TOOLTIP_NAME, props.__scopeTooltip);
+  const ref = React.useRef<TooltipContentHoverableElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+  const [pointerGraceArea, setPointerGraceArea] = React.useState<Polygon | null>(null);
+  const pointerGraceTimerRef = React.useRef(0);
+
+  const { onClose } = context;
+  const content = ref.current;
+  const trigger = context.trigger;
+
+  const handleRemoveGraceArea = React.useCallback(() => {
+    setPointerGraceArea(null);
+    removeDebugArea();
+    window.clearTimeout(pointerGraceTimerRef.current);
+    providerContext.isPointerInTransitRef.current = false;
```

Comment:
I think generally we try to avoid mutating refs directly on context, and instead provide an interface with callbacks.

Reply:
I added a single callback that takes an arg, let me know if this appropriate or if we prefer two.